snapshot-labs/snapshot
# feat(proposal): add ability to flag proposals for admins and moderators

### Feature
[Flag proposals](https://www.notion.so/snapshotlabs/Move-flagged-and-verified-property-to-hub-b6b076290ee34162ac34fa107e99c0cf?pvs=4)

### Changes 
1. Add action under "..." on proposal page

### How to test
1. Open proposal
2. Click on "..."
3. Click "Flag proposal"
<img width="330" alt="Screenshot 2023-09-21 at 13 55 11" src="https://github.com/snapshot-labs/snapshot/assets/7693369/1b6b62a7-3f94-40c2-849a-feeb92926303">

4. Sign message
5. Refresh the page, and observe that the proposal is marked 
<img width="716" alt="Screenshot 2023-09-21 at 13 56 09" src="https://github.com/snapshot-labs/snapshot/assets/7693369/e2c0b90e-c0ab-47e4-b95f-d39ab7a71db0">


### To-Do
- [ ] marge [PR](https://github.com/snapshot-labs/snapshot.js/pull/903) from `snapshot.js`
- [ ] upgrade `@snapshot-labs/snapshot.js` to a new version
- [ ] merge [PR](https://github.com/snapshot-labs/snapshot-hub/pull/679) from `snapshot-hub`
- [ ] merge [PR](https://github.com/snapshot-labs/snapshot-sequencer/pull/168) from `snapshot-sequencer`


### Self-review checklist

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*
